panel.py

```diff
@@ -3,101 +3,106 @@
-  # (this is because the keys of `actions` won't be able to be JSON-serialized if numeric).
-  #
-  # The values of this will depend on the control. For a button or a switch, `state` might be one
-  # of `['0', '1']` or `['off', 'on']`. For a multi-valued control like a dial or slider, `state`
-  # might be one of `['0', '1', '2', '3']` or `['red', 'green', 'yellow'].
-  'state': '0',
-
-  # Possible things that the player might do to or with the control. This is a map where the keys
-  # are the possible values of `state` (see above), and the values are how the player puts the
-  # control in that state ("actions").
-  #
-  # You'll notice that the value of '0' here is the empty string. This represents the player not
-  # having to do anything to put the control in this state, as would be the case for a button-like
-  # control, where if the player stops touching the control it will automatically return to this
-  # state.
-  #
-  # The game may ask the player to perform any non-empty action. The action will be displayed to the
-  # player, so make it clear and exciting. Actions should be globally unique, since players will be
-  # shouting these across the room. It's possible to unique actions by being creative about what
-  # the control is called and does (and labeling the physical control accordingly). For instance,
-  # notice how the action here is to "defenestrates the aristocracy", not merely "push the button".
-  # Another button might "launch the missiles". Two dials might respectively be labeled
-  # "Froomulator" and "Hypervisor", with actions "Set Froomulator to 1" vs. "Set Hypervisor to 1".
-  #
-  # See the third control below for a shorthand way to define `actions` for dial-like controls.
-  'actions': {
-    '0': '',
-    '1': 'Defenestrate the aristocracy!'
-  },
-
-    # HACK(jeff): `'type': 'button'` is only present to be able to play the game via the CLI, see
-    # where this is read at the bottom of this script.
-  'type': 'button',
-}, {
-  'id': 'octo',
-
-  'state': 'nothing',
-
-  'actions': {
-    'nothing': '',
-    'nipple': 'Octo bite raven girl nipple!',
-    'mouth': 'Octo kiss raven girl mouth!'
-  },
-
-  'type': 'button'
-}, {
-  'id': 'Froomulator',
-
-  'state': '0',
-
-  # This is a shorthand form of `states`, where the first value (the array) contains the values for
-  # `state`, and the second value (the string) is a "template action": the actual actions for each
-  # state will be formed by replacing "%s" with the state.
-  'actions': [
-    ['0', '1', '2'],
-    'Set Froomulator to %s!'
+# connect/disconnect, add and remove from this list, then call `announce`
+# again.
+controls = [
+    {
+      # An internal identifier for the control. Must be globally unique. Pick a
+      # name that's meaningful for you, this won't be seen by the player.
+      'id': 'defenestrator',
+
+      # Type -- must be one of 'button', 'switch', or 'selector'
```

Comment:
Can you speak generally about why we need to get much more structured with the control schema? My impression is that the previous schema was sufficient to represent all these control types, while also being more concise.

Refactoring the schema would also require nontrivial server-side changes, not yet in this PR.

I can see how this structure makes validation slightly easier, and that is a nice gesture, but strikes me as a nice-to-have—we can validate just by looking at each other's code / running it. :)